### ISSUE TITLE
ruby2_keywords breaks Style/DocumentationMethod cop

### ISSUE THREAD

**Issue by johnnyshields** [rubocop/rubocop#11586]
`Style/DocumentationMethod` cop will yield a false error for the following:

```ruby
# This is the method doc.
ruby2_keywords def bazqux(name, *args, &block)
end

#=> Style/DocumentationMethod: Missing method documentation comment.
```

Other modifiers such as module_function work correctly:

```ruby
# This is the method doc.
module_function def foobar(name, *args, &block)
end

#=> no Rubocop error
```
